src/runtime/manifest.ts

```diff
@@ -1303,16 +1303,24 @@ ${e.message}
       }
       const params: Dictionary<string|number|boolean|{}> = {};
       for (const param of aRefItem.params) {
-        if (params[param.name]) {
-          throw new ManifestError(
-              aRefItem.location, `annotation '${annotation.name}' already has param '${param.name}'`);
-        }
-        const aParam = annotation.params[param.name];
-        if (!aParam) {
-          throw new ManifestError(
-              aRefItem.location, `unexpected annotation param: '${param.name}'`);
+        if (param.kind === 'annotation-named-param') {
+          if (params[param.name]) {
+            throw new ManifestError(
+                aRefItem.location, `annotation '${annotation.name}' already has param '${param.name}'`);
```

Comment:
Is that a bug? I'm not sure I understand the error message in this context.

Reply:
updated comment, and added test.